jax/interpreters/batching.py

```diff
@@ -621,7 +621,7 @@ def _handle_scalar_broadcasting(nd, x, d):
   if d is not_mapped or nd == np.ndim(x):
     return x
   else:
-    return x.reshape(x.shape + (1,) * (nd - np.ndim(x)))
+    return jax.numpy.expand_dims(x, tuple(range(np.ndim(x), nd)))
```

Comment:
Similar here, let's prefer `lax.expand_dims`. It would be better to avoid dependence on `jax.numpy` within this file.